src/hooks/ContractReader/ContractReadValue.ts

```diff
@@ -28,7 +28,8 @@ export function useContractReadValue<C extends string, V>({
   formatter?: (val?: any) => V | undefined // eslint-disable-line @typescript-eslint/no-explicit-any
   valueDidChange?: (oldVal?: V, newVal?: V) => boolean
 }) {
-  const [value, setValue] = useState<V | undefined>()
+  const fetchId = useRef(0)
```

Comment:
**question**: @wraeth-eth this feels like it might be an anti-pattern. How was this working previously, and can we re-implement that same logic or revert the PR that introduced the bug?

Reply:
I can remove it completely , the bug is solved by the dependency array change below.

I left this in as I thought it might help with stuff, but it isn’t necessary

Reply:
removed